[Quant] [Inductor] Enable the Inductor Lowering of QConv2d post op hardswish

Le-Zheng opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115147
* #115146
* #115145

**Summary**
Enable the fusion pattern of `QConv2d -> hardswish` lowering to `hardswish` as `QConv2d` post operator. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_hardswish_cpu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_qconv2d_hardswish
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Le-Zheng committed:
[Quant] [Inductor] Enable the Inductor Lowering of QConv2d post op hardswish

[ghstack-poisoned]

Le-Zheng committed:
Update on "[Quant] [Inductor] Enable the Inductor Lowering of QConv2d post op hardswish"


**Summary**
Enable the fusion pattern of `QConv2d -> hardswish` lowering to `hardswish` as `QConv2d` post operator. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_hardswish_cpu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_qconv2d_hardswish
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

jgong5 commented on torch/_inductor/fx_passes/quantization.py:
```diff
@@ -117,6 +117,31 @@ def _may_generate_pattern_with_dtype_convert(pattern, dtype=Arg(), dtype_convert
 )
 
+
+def get_dequantize(users=1):
```
Can we extend `dequantize_qconv_pt2e_pattern` to avoid duplicate code?

Le-Zheng replied:
Thank you for your feedback! I made a change to utilze get_dequantize for where using `dequantize_qconv_pt2e_pattern` to avoid the duplicate.

jgong5 replied:
Please give it a meaningful name here, e.g., `get_dequantize_qconv_pt2e_pattern`.

Le-Zheng committed:
Update on "[Quant] [Inductor] Enable the Inductor Lowering of QConv2d post op hardswish"


**Summary**
Enable the fusion pattern of `QConv2d -> hardswish` lowering to `hardswish` as `QConv2d` post operator. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_hardswish_cpu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_qconv2d_hardswish
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Le-Zheng committed:
Update on "[Quant] [Inductor] Enable the Inductor Lowering of QConv2d post op hardswish"


**Summary**
Enable the fusion pattern of `QConv2d -> hardswish` lowering to `hardswish` as `QConv2d` post operator. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_hardswish_cpu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_qconv2d_hardswish
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Le-Zheng committed:
Update on "[Quant] [Inductor] Enable the Inductor Lowering of QConv2d post op hardswish"


**Summary**
Enable the fusion pattern of `QConv2d -> hardswish` lowering to `hardswish` as `QConv2d` post operator. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_hardswish_cpu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_qconv2d_hardswish
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Le-Zheng commented:
@pytorchbot merge

Commit:
Update on "[Quant] [Inductor] Enable the Inductor Lowering of QConv2d post op hardswish"


**Summary**
Enable the fusion pattern of `QConv2d -> hardswish` lowering to `hardswish` as `QConv2d` post operator. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_hardswish_cpu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_qconv2d_hardswish
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Le-Zheng committed:
Update on "[Quant] [Inductor] Enable the Inductor Lowering of QConv2d post op hardswish"


**Summary**
Enable the fusion pattern of `QConv2d -> hardswish` lowering to `hardswish` as `QConv2d` post operator. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_hardswish_cpu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_qconv2d_hardswish
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Le-Zheng committed:
Update on "[Quant] [Inductor] Enable the Inductor Lowering of QConv2d post op hardswish"


**Summary**
Enable the fusion pattern of `QConv2d -> hardswish` lowering to `hardswish` as `QConv2d` post operator. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_hardswish_cpu
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_qconv2d_hardswish
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]
